specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json

```diff
@@ -668,6 +668,50 @@
         }
       }
     },
+    "EncryptionDetails": {
+      "description": "Details of the encryption associated with the workspace",
+      "type": "object",
+      "properties": {
+        "doubleEncryptionEnabled": {
+          "description": "Double Encryption enabled",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "cmk": {
+          "description": "Customer Managed Key Details",
+          "$ref": "#/definitions/CustomerManagedKeyDetails"
+        }
+      }
+    },
+    "CustomerManagedKeyDetails": {
+      "description": "Details of the customer managed key associated with the workspace",
+      "type": "object",
+      "properties": {
+        "status": {
+          "description": "The customer managed key status on the workspace",
+          "type": "string",
+          "readOnly": true
+        },
+        "key": {
+          "description": "The key object of the workspace",
+          "$ref": "#/definitions/WorkspaceKeyDetails"
+        }
+      }
+    },
+    "WorkspaceKeyDetails": {
+      "description": "Details of the customer managed key associated with the workspace",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Workspace Key sub-resource name",
+          "type": "string"
+        },
+        "keyVaultUrl": {
```

Comment:
Either the keys should be managed as child resources or they are managed here.  It shouldn't be a mix of both.  Please remove the keyVaultUrl here and just take the child key resource's ID.  The KeyVault information should live there (and already does)

Reply:
In order to not have a 3-step provisioning phase but rather a 2-step provisioning phase (already unintuitive to the customer, we create a sub-resource in the first PUT workspace call. Then, when the customer does a PUT on the sub-resource that was created through the workspace body, we accept it since we intentionally created it for the user.

Reply:
You have keys modeled as a tracked resource in this spec.  It should not be getting created behind the scenes.  If you want to create it behind the scenes please model it as a proxy resource (without its own tags/location).

Reply:
Changed to proxy resource